rest/route/annotations.go

```diff
@@ -213,3 +216,104 @@ func (h *annotationByTaskHandler) Run(ctx context.Context) gimlet.Responder {
 
 	return gimlet.NewJSONResponse(res)
 }
+
+////////////////////////////////////////////////////////////////////////
+//
+// PUT /rest/v2/task/{task_id}/annotation
+
+type annotationByTaskPutHandler struct {
+	taskId    string
+	execution int
+
+	user       gimlet.User
+	annotation *annotations.TaskAnnotation
+	sc         data.Connector
+}
+
+func makePutAnnotationsByTask(sc data.Connector) gimlet.RouteHandler {
+	return &annotationByTaskPutHandler{
+		sc: sc,
+	}
+}
+
+func (h *annotationByTaskPutHandler) Factory() gimlet.RouteHandler {
+	return &annotationByTaskPutHandler{
+		sc: h.sc,
+	}
+}
+
+func (h *annotationByTaskPutHandler) Parse(ctx context.Context, r *http.Request) error {
+	var err error
+
+	h.taskId = gimlet.GetVars(r)["task_id"]
+	if h.taskId == "" {
+		return gimlet.ErrorResponse{
+			Message:    "task ID cannot be empty",
+			StatusCode: http.StatusBadRequest,
+		}
+	}
+	execution := r.URL.Query().Get("execution")
+	if execution != "" {
+		h.execution, err = strconv.Atoi(execution)
+		if err != nil {
+			return gimlet.ErrorResponse{
+				Message:    fmt.Sprintf("Invalid execution: '%s'", err.Error()),
+				StatusCode: http.StatusBadRequest,
+			}
+		}
+	} else {
+		h.execution = -1
+	}
+
+	body := util.NewRequestReader(r)
+	defer body.Close()
+	if err = json.NewDecoder(body).Decode(&h.annotation); err != nil {
+		return gimlet.ErrorResponse{
+			Message:    fmt.Sprintf("API error while unmarshalling JSON: '%s'", err.Error()),
+			StatusCode: http.StatusBadRequest,
+		}
+	}
+
+	if h.annotation.TaskId == "" {
+		return gimlet.ErrorResponse{
+			Message:    "TaskID is required",
+			StatusCode: http.StatusBadRequest,
+		}
+	}
+
+	// check if the task exists
+	t, err := task.FindOne(task.ById(h.taskId))
+	if err != nil {
+		return err
+	}
+	if t == nil {
+		return gimlet.ErrorResponse{
+			Message:    fmt.Sprintf("the task %s does not exist", h.taskId),
+			StatusCode: http.StatusBadRequest,
+		}
+	}
+
+	u := MustHaveUser(ctx)
+	h.user = u
+	return nil
+}
+
+func (h *annotationByTaskPutHandler) Run(ctx context.Context) gimlet.Responder {
+	a := h.annotation
+	source := &annotations.Source{
+		Author:    h.user.DisplayName(),
+		Time:      time.Now(),
+		Requester: annotations.APIRequester,
+	}
+
+	err := annotations.UpdateAnnotation(h.taskId, h.execution, a, source)
```

Comment:
you can just pass h.annotation in directly here; you also don't need to explicitly pass in h.taskId or h.execution since they should be the same as the ones in the annotation